### PR TITLE
[Feature] Video grid orientation

### PR DESCRIPTION
**Pull request by David-Henner**
## What's new in this PR?

https://wearezeta.atlassian.net/browse/ZIOS-13479

Match video tiles orientation to device orientation.

### Notes

This draft covers the following elements needed to support the orientation changes:
1. Listening to device orientation events
2. Rotating the individual video tiles according to device orientation
3. Adapting the user details constraints to the new orientation 

### TODO

- [x] Support iPad scenario
- [x] Unit tests

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Calling/VideoGridView/BaseVideoPreviewView.swift [hunk 1]
```diff
@@ -114,3 +127,44 @@ class BaseVideoPreviewView: UIView, AVSIdentifierProvider {
         })
     }
 }
+
+// MARK: - User Details Constraints
+private struct UserDetailsConstraints {
+    private var bottom: NSLayoutConstraint
+    private var leading: NSLayoutConstraint
+    private var trailing: NSLayoutConstraint
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/Calling/VideoGridView/BaseVideoPreviewView.swift:135]: can we use `let`?

File: Wire-iOS/Sources/UserInterface/Helpers/OrientationDelta.swift [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            return 0
+        }
+    }
+}
+
+enum OrientationAngle {
+    case none // 0°
+    case right // 90°
+    case straight // 180°
+    case full // 360°
```
Comment (marcomaddy1982) [inline Wire-iOS/Sources/UserInterface/Helpers/OrientationDelta.swift:82]: is 0 not equal to 360? :-D

File: Wire-iOS/Sources/UserInterface/Helpers/OrientationDelta.swift [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        case OrientationAngle.right.radians:
+            self = .rotatedLeft
+        case -OrientationAngle.right.radians:
+            self = .rotatedRight
+        case OrientationAngle.none.radians:
+            self = .equal
+        default:
+            self = .unknown
+        }
+    }
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/Helpers/OrientationDelta.swift:49]: Beware of comparing two floating point numbers, the default implementation of equality is typically exact equality, and floating point arithmetic isn't exact. Perhaps you could implement this in a more robust way?

Comment (David-Henner) [reply Wire-iOS/Sources/UserInterface/Helpers/OrientationDelta.swift]: Updated to something more robust 5258505

File: Wire-iOS/Sources/Helpers/Array+Shift.swift [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+extension Array {
+    
+    /// Shifts the array by the given amount
+    /// Negatives shift left, positives shift right
+    ///
+    /// [1, 2, 3]
+    /// shifted by 1 => [3, 1, 2]
+    /// shifted by -1 => [2, 3, 1]
+    func shifted(by amount: Int) -> Array {
```
[the remark this answers is not in the document]
Comment (David-Henner) [reply Wire-iOS/Sources/Helpers/Array+Shift.swift]: done here 517d877

File: Wire-iOS/Sources/UserInterface/Calling/VideoGridView/BaseVideoPreviewView.swift (hunk 1 above)
Comment (David-Henner) [reply Wire-iOS/Sources/UserInterface/Calling/VideoGridView/BaseVideoPreviewView.swift]: done ac95a27

File: Wire-iOS/Sources/UserInterface/Helpers/OrientationDelta.swift (hunk 2 above)
Comment (David-Henner) [reply Wire-iOS/Sources/UserInterface/Helpers/OrientationDelta.swift]: Yes 😄 Removed.

**Review by johnxnguyen — APPROVED** [review]

**Pull request merged**